app/src/main/java/com/example/dndwiki/repository/SpellRepository.kt

```diff
@@ -36,13 +32,9 @@ class SpellRepository(
         return spellDetails
     }
 
-    suspend fun saveAllSpellDetails() {
-        if (!alreadyExecuted) {
-            api.getSpells().forEach {
-                db.saveSpellDetails(api
-                    .getSpellDetails(it.index))
-            }
-            alreadyExecuted = true
+    private suspend fun saveAllSpellDetails() {
```

Comment:
https://github.com/RoySuttieKLM/DnDWiki/issues/30

To fix in a different PR :)